plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -0,0 +1,270 @@
+
+            ImmutableList.Builder<CompletableFuture<GetColumnStatisticsForTableResult>> getStatsFutures = ImmutableList.builder();
+
+            columnChunks.forEach(partialColumns -> getStatsFutures.add(CompletableFuture.supplyAsync(() -> {
+                final GetColumnStatisticsForTableRequest request = new GetColumnStatisticsForTableRequest()
+                        .withCatalogId(catalogId)
+                        .withDatabaseName(table.getDatabaseName())
+                        .withTableName(table.getTableName())
+                        .withColumnNames(partialColumns);
+
+                return glueClient.getColumnStatisticsForTable(request);
+            }, readExecutor)));
+
+            final HiveBasicStatistics tableStatistics = getHiveBasicStatistics(table.getParameters());
+            final ImmutableMap.Builder<String, HiveColumnStatistics> columnStatsMapBuilder = ImmutableMap.builder();
+            for (CompletableFuture<GetColumnStatisticsForTableResult> future : getStatsFutures.build()) {
+                final GetColumnStatisticsForTableResult tableColumnsStats = MoreFutures.getFutureValue(future, TrinoException.class);
+                tableColumnsStats.getColumnStatisticsList()
+                        .forEach(col -> columnStatsMapBuilder.put(col.getColumnName(), GlueStatConverter.fromGlueColumnStatistics(col.getStatisticsData(), tableStatistics.getRowCount())));
+            }
+
+            return columnStatsMapBuilder.build();
+        }
+        catch (AmazonServiceException ex) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
+        }
+    }
+
+    @Override
+    public Map<String, HiveColumnStatistics> getPartitionColumnStatistics(Partition partition)
+    {
+        try {
+            final List<List<Column>> columnChunks = Lists.partition(partition.getColumns(), GLUE_COLUMN_READ_STAT_PAGE_SIZE);
+            ImmutableList.Builder<CompletableFuture<GetColumnStatisticsForPartitionResult>> getStatsFutures = ImmutableList.builder();
+
+            columnChunks.forEach(partialColumns -> getStatsFutures.add(CompletableFuture.supplyAsync(() -> {
+                final List<String> columnsNames = partialColumns.stream()
+                        .map(Column::getName)
+                        .collect(toImmutableList());
+
+                final GetColumnStatisticsForPartitionRequest request = new GetColumnStatisticsForPartitionRequest()
+                        .withCatalogId(catalogId)
+                        .withDatabaseName(partition.getDatabaseName())
+                        .withTableName(partition.getTableName())
+                        .withColumnNames(columnsNames)
+                        .withPartitionValues(partition.getValues());
+
+                return glueClient.getColumnStatisticsForPartition(request);
+            }, readExecutor)));
+
+            final HiveBasicStatistics tableStatistics = getHiveBasicStatistics(partition.getParameters());
```

Comment:
Please restructure in similar way as above. I think it make the code more readable.